Analysis/Tasks/PWGCF/o2femtodream/include/o2femtodream/FemtoDreamTrackSelection.h

```diff
@@ -0,0 +1,480 @@
+
+  const static float pTMin = getMinimalSelection(femtoDreamTrackSelection::kpTMin, femtoDreamSelection::kLowerLimit);
+  const static float pTMax = getMinimalSelection(femtoDreamTrackSelection::kpTMax, femtoDreamSelection::kUpperLimit);
+  const static float etaMax = getMinimalSelection(femtoDreamTrackSelection::kEtaMax, femtoDreamSelection::kAbsUpperLimit);
+  const static float nClsMin = getMinimalSelection(femtoDreamTrackSelection::kTPCnClsMin, femtoDreamSelection::kLowerLimit);
+  const static float fClsMin = getMinimalSelection(femtoDreamTrackSelection::kTPCfClsMin, femtoDreamSelection::kLowerLimit);
+  const static float cTPCMin = getMinimalSelection(femtoDreamTrackSelection::kTPCcRowsMin, femtoDreamSelection::kLowerLimit);
+  const static float sClsMax = getMinimalSelection(femtoDreamTrackSelection::kTPCsClsMax, femtoDreamSelection::kLowerLimit);
+  const static float dcaxyMax = getMinimalSelection(femtoDreamTrackSelection::kDCAxyMax, femtoDreamSelection::kAbsUpperLimit);
+  const static float dcazMax = getMinimalSelection(femtoDreamTrackSelection::kDCAzMax, femtoDreamSelection::kAbsUpperLimit);
+  const static float nSigmaPIDMax = getMinimalSelection(femtoDreamTrackSelection::kPIDnSigmaMax, femtoDreamSelection::kAbsUpperLimit);
+
+  if (nPtMinSel > 0 && pT < pTMin) {
+    return 0;
+  }
+  if (nPtMaxSel > 0 && pT > pTMax) {
+    return 0;
+  }
+  if (nEtaMaxSel > 0 && std::abs(eta) > etaMax) {
+    return 0;
+  }
+  if (nTPCnMinSel > 0 && tpcNClsF < nClsMin) {
+    return 0;
+  }
+  if (nTPCfMinSel > 0 && tpcRClsC < fClsMin) {
+    return 0;
+  }
+  if (nTPCcMinSel > 0 && tpcNClsC < cTPCMin) {
+    return 0;
+  }
+  if (nTPCsMaxSel > 0 && tpcNClsS > sClsMax) {
+    return 0;
+  }
+  if (nDCAxyMaxSel > 0 && std::abs(dcaXY) > dcaxyMax) {
+    return 0;
+  }
+  if (nDCAzMaxSel > 0 && std::abs(dcaZ) > dcazMax) {
+    return 0;
+  }
+  if (nPIDnSigmaSel > 0) {
+    if (std::all_of(pidTPC.begin(), pidTPC.end(), [](float pid) { return std::abs(pid) > nSigmaPIDMax; }) && std::all_of(pidTOF.begin(), pidTOF.end(), [](float pid) { return std::abs(pid) < nSigmaPIDMax; })) {
+      return 0;
+    }
+  }
+
+  // now iterate over all selections and check whether these are fulfilled
+  float observable;
+  for (auto& sel : mSelections) {
+    const auto selVariable = sel.getSelectionVariable();
+    if (selVariable == femtoDreamTrackSelection::kPIDnSigmaMax) {
+      /// PID needs to be handled a bit differently since we may need more than one species
+      for (size_t i = 0; i < pidTPC.size(); ++i) {
+        auto pidTPCVal = pidTPC.at(i);
+        auto pidTOFVal = pidTOF.at(i);
+        sel.checkSelectionSetBit(pidTPCVal, output, counter);
+        auto pidComb = std::sqrt(pidTPCVal * pidTPCVal + pidTOFVal * pidTOFVal);
+        sel.checkSelectionSetBit(pidComb, output, counter);
+      }
+
+    } else {
+      /// for the rest it's all the same
+      switch (selVariable) {
```

Comment:
Same here. I think all these cases will slow down processing a lot... We can profile this.

Reply:
yes if you have an idea how to improve this that would be very good, since this is one of the core pieces of the generic cut selection we set up (and that in principle could be used by anyone interested)